src/libraries/System.Net.Requests/src/System/Net/HttpWebResponse.cs

```diff
@@ -337,7 +337,36 @@ public override Stream GetResponseStream()
             CheckDisposed();
             if (_httpResponseMessage.Content != null)
             {
-                return _httpResponseMessage.Content.ReadAsStream();
+                Stream contentStream = _httpResponseMessage.Content.ReadAsStream();
+                int maxErrorResponseLength = HttpWebRequest.DefaultMaximumErrorResponseLength;
+                if (maxErrorResponseLength < 0 || StatusCode < HttpStatusCode.BadRequest)
+                {
+                    return contentStream;
+                }
+
+                MemoryStream memoryStream = new MemoryStream();
+                byte[] buffer = new byte[1024];
+                int readLength = 0;
+
+                while (readLength < maxErrorResponseLength)
+                {
+                    int len = contentStream.Read(buffer, 0, Math.Min(maxErrorResponseLength - readLength, buffer.Length));
```

Comment:
This is forcing sync IO even on otherwise async callers.

We may want to move this into a custom stream that truncates while reading instead of buffering upfront.